components/Chips/src/MDCAlignedCollectionViewFlowLayout.h

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+typedef enum : NSUInteger {
+  MDCCollectionHorizontalAlignmentLeading,
+  MDCCollectionHorizontalAlignmentTrailing,
+  MDCCollectionHorizontalAlignmentLeft,
+  MDCCollectionHorizontalAlignmentRight
+} MDCCollectionHorizontalAlignment;
+
+/*
+ MDCAlignedCollectionViewFlowLayout is an aligned version of UICollectionViewFlowLayout.
+
+ Use exactly as you would a standard UICollectionViewLayout. Set minimumInteritemSpacing to control
+ spacing between cells.
+ */
+@interface MDCAlignedCollectionViewFlowLayout : UICollectionViewFlowLayout
```

Comment:
Can we move this flow layout to its own component? This would allow this component to focus solely on representation of the chip view, with clients being able to choose how they integrate it in their app.

We'll probably still recommend using collection views, in which case we'd point them to this layout component.

This also allows clients to make use of this aligned flow layout without pulling in the chip componentry.

Reply:
We should eventually move the layout to its own component. I think the best plan of attack might be to make this MDCChipCollectionViewLayout, and then build out a more full featured and generic Aligned layout later.